File: lib/model/transaction_events.dart
enum TransactionEvents {
  unknown,
  transactionConfirm,
  transactionFailed,
  transactionOk,
  transactionError
}

extension TransactionEventsString on String  {
  TransactionEvents get transactionEvent {
    switch (this) {
      case 'TRANSACTION_CONFIRM':
        return TransactionEvents.transactionConfirm;
      case 'TRANSACTION_FAILED':
        return TransactionEvents.transactionFailed;
      case 'TRANSACTION_OK':
        return TransactionEvents.transactionOk;
      case 'TRANSACTION_ERROR':
        return TransactionEvents.transactionError;
    }
    return TransactionEvents.unknown;
  }
}

File: android/src/main/kotlin/com/paulohbsousa/clisitef/TransactionEvents.kt
package com.paulohbsousa.clisitef

enum class TransactionEvents(val named: String) {
    TRANSACTION_CONFIRM("TRANSACTION_CONFIRM"),
    TRANSACTION_FAILED("TRANSACTION_FAILED"),
    TRANSACTION_OK("TRANSACTION_OK"),
    TRANSACTION_ERROR("TRANSACTION_ERROR")
}
